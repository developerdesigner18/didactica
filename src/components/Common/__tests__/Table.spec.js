import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import LxpTable from "../Table.vue";
import LxpCheckbox from "../Checkbox.vue";
import LxpPagination from "../Pagination.vue";

import { BPagination, BFormCheckbox } from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-table", LxpTable);
localVue.component("lxp-checkbox", LxpCheckbox);
localVue.component("lxp-pagination", LxpPagination);
localVue.component("b-pagination", BPagination);
localVue.component("b-form-checkbox", BFormCheckbox);

describe("Table.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should show the sort based on tableHeader `sortable` prop", async () => {
      wrapper = await getWrapper();
      await waitRAF();
      const tableVariant = wrapper.find(".lxp-table__sort");
      expect(tableVariant.exists()).toBe(true);
    });

    it("should update the variant based on prop", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({ variant: "success" });
      await waitRAF();
      const tableVariant = wrapper.find(".table-success");
      expect(tableVariant.exists()).toBe(true);
    });

    it("should update the checkbox when `selectable` is true", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({ selectable: true });
      await waitRAF();
      const checkbox = wrapper.find(".lxp-checkbox");
      expect(checkbox.exists()).toBe(true);
    });

    it("should show pagination when `hasPagination` is true", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({ hasPagination: true });
      await waitRAF();
      const checkbox = wrapper.find(".lxp-pagination");
      expect(checkbox.exists()).toBe(true);
    });
  });

  describe("Events", () => {
    let wrapper;

    beforeEach(async () => {
      wrapper = await getWrapper();
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it("should emit `sort` when sort button is clicked", async () => {
      let sortableBtn = wrapper.find(".lxp-table__sort");
      await sortableBtn.trigger("click");
      expect(wrapper.emitted().sort).toBeTruthy();
    });
  });
});

function getWrapper() {
  let wrapper = mount(LxpTable, {
    propsData: {
      tableHeader: [
        {
          key: "age",
          name: "Age",
          sortable: true
        },
        {
          key: "first_name",
          name: "First Name",
          sortable: false
        },
        {
          key: "last_name",
          name: `Last name`,
          sortable: true
        }
      ],
      tableData: [
        { age: 40, first_name: "Dickerson", last_name: "Macdonald" },
        { age: 21, first_name: "Larsen", last_name: "Shaw" },
        { age: 89, first_name: "Geneva", last_name: `<b>Wilson</b>` }
      ]
    },
    localVue
  });

  return wrapper;
}
