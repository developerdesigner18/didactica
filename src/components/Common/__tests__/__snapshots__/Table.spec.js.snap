// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.vue render when created  1`] = `
<div
  class="lxp-table__wrapper"
>
  <table
    class="lxp-table table-primary"
  >
    <thead>
      <tr>
        <!---->
         
        <th>
          <div
            class="lxp-table__heading"
          >
            
            Age
            
            <span
              class="lxp-table__sort"
            >
              <img
                alt="sort"
                height="24"
                loading="lazy"
                src="https://files.lxp.academy.who.int/didactica/assets/images/table/sort-down.svg"
                width="24"
              />
            </span>
          </div>
        </th>
        <th>
          <div
            class="lxp-table__heading"
          >
            
            First Name
            
            <!---->
          </div>
        </th>
        <th>
          <div
            class="lxp-table__heading"
          >
            
            Last name
            
            <span
              class="lxp-table__sort"
            >
              <img
                alt="sort"
                height="24"
                loading="lazy"
                src="https://files.lxp.academy.who.int/didactica/assets/images/table/sort-down.svg"
                width="24"
              />
            </span>
          </div>
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <!---->
         
        <td>
          <div>
            40
          </div>
        </td>
        <td>
          <div>
            Dickerson
          </div>
        </td>
        <td>
          <div>
            Macdonald
          </div>
        </td>
      </tr>
      <tr>
        <!---->
         
        <td>
          <div>
            21
          </div>
        </td>
        <td>
          <div>
            Larsen
          </div>
        </td>
        <td>
          <div>
            Shaw
          </div>
        </td>
      </tr>
      <tr>
        <!---->
         
        <td>
          <div>
            89
          </div>
        </td>
        <td>
          <div>
            Geneva
          </div>
        </td>
        <td>
          <div>
            <b>
              Wilson
            </b>
          </div>
        </td>
      </tr>
    </tbody>
     
    <tfoot>
      <tr>
        <div
          class="ltw__footer"
        >
          <div
            class="ltw-footer__left"
          />
           
          <!---->
        </div>
      </tr>
    </tfoot>
  </table>
</div>
`;
